versions/1.17.1-fabric/src/main/kotlin/gg/essential/universal/shader/MCShader.kt

```diff
@@ -173,21 +175,29 @@ internal class ShaderTransformer {
             replacements["gl_Color"] = "uc_FrontColor"
         }
 
-        fun replaceAttribute(needle: String, type: String, replacementName: String = "uc_" + needle.substringAfter("_"), replacement: String = replacementName) {
+        fun replaceAttribute(newAttributes: MutableList<Pair<String, String>>, needle: String, type: String, replacementName: String = "uc_" + needle.substringAfter("_"), replacement: String = replacementName) {
             if (needle in source) {
                 replacements[needle] = replacement
                 if (replacementName !in attributes) {
                     attributes.add(replacementName)
-                    transformed.add("in $type $replacementName;")
+                    newAttributes.add(replacementName to "in $type $replacementName;")
                 }
             }
         }
         if (vert) {
-            replaceAttribute("gl_Vertex", "vec3", replacement = "vec4(uc_Vertex, 1.0)")
-            replaceAttribute("gl_Color", "vec4")
-            replaceAttribute("gl_MultiTexCoord0.st", "vec2", "uc_UV0")
-            replaceAttribute("gl_MultiTexCoord1.st", "vec2", "uc_UV1")
-            replaceAttribute("gl_MultiTexCoord2.st", "vec2", "uc_UV2")
+            val newAttributes = mutableListOf<Pair<String, String>>()
+            replaceAttribute(newAttributes, "gl_Vertex", "vec3", replacement = "vec4(uc_Vertex, 1.0)")
+            replaceAttribute(newAttributes, "gl_Color", "vec4")
+            replaceAttribute(newAttributes, "gl_MultiTexCoord0.st", "vec2", "uc_UV0")
+            replaceAttribute(newAttributes, "gl_MultiTexCoord1.st", "vec2", "uc_UV1")
+            replaceAttribute(newAttributes, "gl_MultiTexCoord2.st", "vec2", "uc_UV2")
+
+            if (vertexFormat != null) {
+                newAttributes.sortedBy { vertexFormat.mc.shaderAttributes.indexOf(it.first.removePrefix("uc_")) }
```

Comment:
Works for all but `uc_Vertex` which MC calls `Position`.
(works in practice anyway because all standard MC formats currently have Position first, so -1 for the index gets it sorted properly too)

Reply:
should I just change `uc_Vertex` to `uc_Position` to be consistent with the others?

Reply:
sgtm